clients/js/src/plugins/types.ts

```diff
@@ -0,0 +1,134 @@
+import {
+  BurnDelegate,
+  FreezeDelegate,
+  PermanentFreezeDelegate,
+  TransferDelegate,
+  UpdateDelegate,
+  Attributes,
+  PermanentTransferDelegate,
+  PermanentBurnDelegate,
+  Edition,
+  basePluginAuthority as pluginAuthority,
+  baseUpdateAuthority as updateAuthority,
+  baseRuleSet as ruleSet,
+  FreezeDelegateArgs,
+  UpdateDelegateArgs,
+  AttributesArgs,
+  PermanentFreezeDelegateArgs,
+  EditionArgs,
+  BasePluginAuthority,
+  BaseRoyaltiesArgs,
+} from '../generated';
+import { RoyaltiesArgs, RoyaltiesPlugin } from './royalties';
+import { PluginAuthority } from './pluginAuthority';
+
+// for backwards compatibility
+export { pluginAuthority, updateAuthority, ruleSet };
+
+export type BasePlugin = {
+  authority: PluginAuthority;
+  offset?: bigint;
+};
```

Comment:
So in general `BaseXyz` now seems to be autogenerated via Kinobi, and then we add some kind of decorated/alternate behavior on top of it and then it becomes just `Xyz`.

But in the case of `BasePlugin` I think it means something slightly different, a more object-oriented meaning that it's kind of a "base class" then extra plugin data is added for different kinds of plugins.

Is that correct understanding, and if so, do we like that kind of terminology given that they are slightly different usage?

Perhaps it is fine because no matter who generates and defines the types, there's a "base class" behavior and then there'san extended behavior beyond the base class behavior.